app/controllers/settings/apps/unsplash.js

```diff
@@ -9,15 +9,24 @@ export default Controller.extend({
     config: injectService(),
     unsplash: injectService(),
 
-    model: alias('settings.unsplash.firstObject'),
+    model: alias('settings.unsplash'),
     testRequestDisabled: empty('model.applicationId'),
 
     save: task(function* () {
         let unsplash = this.get('model');
         let settings = this.get('settings');
+        let hasValidated = unsplash.get('hasValidated');
+        console.log(this.get('model'));
+
+        if (this.get('config.unsplashAPI')) {
+            hasValidated.pushObject('applicationId');
```

Comment:
@AileenCGN I changed this conditional to the following:

```javascript
        // api key field is hidden if set via config so don't validate in that case
        if (!this.get('config.unsplashAPI')) {
            try {
                yield unsplash.validate({property: 'applicationId'});
            } catch (error) {
                console.log('validation failed', error);
                // validation failed, quit early
                if (error === undefined) {
                    return;
                }

                throw error;
            }
        }
```

The `console.log` in this case returned:

```
validation failed ["The validator specified, "unsplashIntegration", did not exist!"]
```

This points to the fact that the validation hasn't been registered in [validation-engine.js](https://github.com/AileenCGN/Ghost-Admin/blob/cc289167816a8943204bbecaa0d5cceda4205518/app/mixins/validation-engine.js#L37-L50)

Hopefully that points you in the right direction 😄